src/main/java/core/basesyntax/service/implementation/ShopServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.service.implementation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.BalanceHandler;
+import core.basesyntax.strategy.PurchaseHandler;
+import core.basesyntax.strategy.ReturnHandler;
+import core.basesyntax.strategy.SupplyHandler;
+
+public class ShopServiceImpl implements ShopService {
+    private Storage storage;
+
+    public ShopServiceImpl() {
+    }
+
+    public ShopServiceImpl(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void process(FruitTransaction transaction) {
```

Comment:
I would prefere to create some class for example StrategyImpl wich will hold Map<Operation, OperationHandler> and in this class just get some operationHandler from this map, and execute operation

Reply:
done